pkg/providers/amifamily/suite_test.go

```diff
@@ -140,6 +140,16 @@ var _ = Describe("AMIProvider", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(amis).To(HaveLen(4))
 	})
+	It("should succeed to resolve AMIs (AL2023)", func() {
+		nodeClass.Spec.AMIFamily = &v1beta1.AMIFamilyAL2023
+		awsEnv.SSMAPI.Parameters = map[string]string{
+			fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2023/x86_64/standard/recommended/image_id", version): amd64AMI,
```

Comment:
can we rely on the default AMIs function here?

Reply:
We could grab the SSM parameters from there but we would still have to manually assign the AMIs to the matching parameters so I'm not sure it actually makes it any simpler. Plus, this is an extra layer of protection for if we accidentally deleted an SSM parameter. Not like that should happen through code review but some defense in depth.